### PR TITLE
app metrics

### PR DESCRIPTION
**Pull request by brindy**
Task/Issue URL: 

* Implement pixel request: https://app.asana.com/0/392891325557410/738818866019760
* Grafana parameters: https://app.asana.com/0/392891325557410/751512552599843

Tech Design URL: https://app.asana.com/0/392891325557410/726260612891763
CC:

**Description**:

* When user launches the app, executes forget all, or opens the privacy dashboard a pixel request is triggered.
* A pixel request is in the following format https://improving.duckduckgo.com/t/{pixel}_{platform}_{form_factor}?atb={atb}
  * atb: a valid atb or blank for the very first launch
  * platform: `android`
  * form factor: `phone` or `tablet` where the device is a tablet if its longest edge >= 600dp

**Steps to test this PR**:

***Pixel requests***:
1. Open kibana and use this query: `(pixel_id:ml) OR (pixel_id:mf) OR (pixel_id:mp)`
1. Launch the app, wait a few seconds and run the query, the ml request should appear
1. Perform a forget all while browsing or from the home screen, wait a few seconds and run the query, the mf request should appear
1. Perform a forget all in the tab switcher, wait a few seconds and run the query, the mf request should appear
1. Open the privacy dashboard, wait a few seconds and run the query, the mf request should appear

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Comment by brindy** [comment]
@CDRussell can you take a quick look at this, please - just want to check the general approach is ok. I'll be looking for a detailed review once I've done the opt out/survey tasks. Thanks!

File: app/src/main/java/com/duckduckgo/app/di/StatisticsModule.kt [hunk 1]
```diff
@@ -16,25 +16,51 @@
...
+    fun statisticsService(retrofit: Retrofit): StatisticsService = retrofit.create(StatisticsService::class.java)
 
     @Provides
     fun statisticsUpdater(statisticsDataStore: StatisticsDataStore, statisticsService: StatisticsService, variantManager: VariantManager) : StatisticsUpdater =
         StatisticsRequester(statisticsDataStore, statisticsService, variantManager)
 
+    @Provides
+    fun pixelService(apiRequestInterceptor: ApiRequestInterceptor) : PixelService {
+
+        // ideally these would be injected but @Named parameters seem to break dagger
```
Comment (CDRussell) [inline app/src/main/java/com/duckduckgo/app/di/StatisticsModule.kt]: Curious to know more about this - `@Named` should work ok

Comment (brindy) [reply app/src/main/java/com/duckduckgo/app/di/StatisticsModule.kt]: Thanks @CDRussell ! I took a second look and got it to work :)

The error message I was getting wasn't clear, it was complaining about something not being marked as `@Nullable` along with a bunch of other Dagger guff, but the actual error was that I hadn't annotated every place the named instances were being injected.

**Comment by brindy** [comment]
@CDRussell - this is now ready to review.  Thanks!

**Comment by CDRussell** [comment]
On the privacy dashboard, if you flip the phone, it sends another pixel.

File: app/src/main/java/com/duckduckgo/app/statistics/api/PixelService.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import com.duckduckgo.app.global.AppUrl
+import io.reactivex.Completable
+import retrofit2.http.GET
+import retrofit2.http.Path
+import retrofit2.http.Query
+
+interface PixelService {
+
+    // We don't have tablet specific UI at all on Android yet, so form factor is irrelevant, thus default to phone
```
Comment (CDRussell) [inline app/src/main/java/com/duckduckgo/app/statistics/api/PixelService.kt]: comment no longer correct?

**Comment by brindy** [comment]
@CDRussell gonna keep the launch pixel as you might not be looking at ATBs at the time and it'll be one off - which may not make much difference but it'll still be wrong.

Have moved the pixel firing in to the view model for the privacy dashboard.

Have fixed formatting in files I've touched.

**Pull request merged**